test/Hanoi.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE KindSignatures     #-}
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE PolyKinds          #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE FlexibleInstances  #-}
+
+------------------------------------------------------------------------
+
+-- Solution to the famous Tower of Hanoi puzzle using tools for state
+-- machine property-based testing.
+--
+-- The puzzle is to move N discs of different sizes from one peg to
+-- another, with one auxiliary peg and a restriction that no disc may ever
+-- be placed on top of a smaller disc. Only one disc can be moved at a time.
+
+------------------------------------------------------------------------
+
+module Hanoi
+  ( Command(..)
+  , Model(..)
+  , initModel
+  , transitions
+  , prop_hanoi
+  ) where
+
+import Data.Kind
+         (Type)
+import Data.Array
+import Data.Maybe
+import Data.TreeDiff.Expr ()
+import GHC.Generics
+         (Generic, Generic1)
+import Prelude
+import Test.QuickCheck
+         (Gen, Property, (===), Arbitrary (arbitrary), choose, suchThat)
+import Test.QuickCheck.Monadic
+         (monadicIO)
+import Test.StateMachine
```

Comment:
Newline here to separate imports of external packages from internal imports (this package, i.e. Test.StateMachine).